Dalamud/Game/Gui/NamePlate/NamePlateInfoView.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Dalamud.Game.Text.SeStringHandling;
+
+using FFXIVClientStructs.FFXIV.Client.UI;
+
+namespace Dalamud.Game.Gui.NamePlate;
+
+/// <summary>
+/// Provides a read-only view of the nameplate info object data for a nameplate. Modifications to
+/// <see cref="NamePlateUpdateHandler"/> fields do not affect this data.
+/// </summary>
+public interface INamePlateInfoView
+{
+    /// <summary>
+    /// Gets the displayed name for this nameplate according to the nameplate info object.
+    /// </summary>
+    SeString Name { get; }
+
+    /// <summary>
+    /// Gets the displayed free company tag for this nameplate according to the nameplate info object. For this field,
+    /// the quote characters which appear on either side of the title are NOT included.
+    /// </summary>
+    SeString FreeCompanyTag { get; }
+
+    /// <summary>
+    /// Gets the displayed free company tag for this nameplate according to the nameplate info object. For this field,
+    /// the quote characters which appear on either side of the title ARE included.
+    /// </summary>
+    SeString QuotedFreeCompanyTag { get; }
+
+    /// <summary>
+    /// Gets the displayed title for this nameplate according to the nameplate info object. For this field, the quote
+    /// characters which appear on either side of the title are NOT included.
+    /// </summary>
+    SeString Title { get; }
+
+    /// <summary>
+    /// Gets the displayed title for this nameplate according to the nameplate info object. For this field, the quote
+    /// characters which appear on either side of the title ARE included.
+    /// </summary>
+    SeString QuotedTitle { get; }
+
+    /// <summary>
+    /// Gets the displayed level text for this nameplate according to the nameplate info object.
+    /// </summary>
+    SeString LevelText { get; }
+
+    /// <summary>
+    /// Gets the flags for this nameplate according to the nameplate info object.
+    /// </summary>
+    int Flags { get; }
```

Comment:
How do you know what these mean? Does it make sense to expose the raw bitset like this as an int, without describing the actual value? (same for the other flags exposed like this)

Reply:
I was a little conflicted on this. The issue is that I believe they aren't fully understood, so I found it hard to have a "proper" enum here. But I also think it's possible to do some things like maybe find the title number from here for example (maybe? I kind of forget myself) and maybe it could be improved over time as we learn more if we let people poke in these flag fields. Other flags can do useful things but it's often the case that I can only explain maybe one or two flags and others are unknown.

We could hide them and just assume people who need them will cast the address to a CS struct though. But it becomes  a bit harder to use then. So not sure what is best here.